### DISCUSSION TITLE
How to add Validation support ?

### DISCUSSION THREAD

**Discussion by ferrisyang** [Netflix/dgs-framework#379] [Q&A]
I try to add validation framework support with springboot-validation. Found that scalar parameter work fine, but java object validation not, because I use codegen plugin, and it will re-generate java type object when start project.
How to add validation support ?

**Comment by hantsy** [comment]
1. use https://github.com/graphql-java/graphql-java-extended-validation to apply validation on graphql schema instead of the object type.
2. For generated types by codegen plugin, use apache commons-lang3 to check the values manually. https://github.com/hantsy/spring-graphql-sample/blob/master/dgs-codegen/src/main/java/com/example/demo/service/PostService.java#L66

**Comment by WtfJoke** [comment]
I made a sample repo, where validations are applied as directives directly in the schema: https://github.com/WtfJoke/graphql-dgs-directives-demo using https://github.com/graphql-java/graphql-java-extended-validation.

Because initially I didnt found a way how to marry/register them in Dgs so I wanted to share my sample repo, which does exactly that :)
